smac/optimizer/smbo.py

```diff
@@ -347,3 +342,25 @@ def _get_timebound_for_intensification(self, time_spent):
                 time_spent, (1-frac_intensify), time_left, frac_intensify))
         return time_left
 
+
+class ChallengerList(object):
+
+    def __init__(self, challengers, configuration_space):
+        self.challengers = challengers
+        self.configuration_space = configuration_space
+        self._index = 0
+        self._next_is_random = False
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._index == len(self.challengers):
+            raise StopIteration
+        elif self._next_is_random:
```

Comment:
I see your point. But as far as I know, we don't have any (empirical) evidence that random configurations are necessary at all. So, I don't know whether it is really important that each intensify call runs at least one random configuration. 
Of course, I don't want to change the default behavior of SMAC. But for the HPOlib benchmarks from Aaron, I would expect that we need a lower frequency of random configurations anyway.